Class refactor

tsohleDev opened the pull request:
refactor code to use classes instead of vanilla objects

tsohleDev committed:
refactor javascript to use classes

tsohleDev committed:
add style

tsohleDev committed:
fix linters

FrederickMih commented on script.js:
```diff
@@ -0,0 +1,98 @@
+const Bookscontainer = document.querySelector('.books');
+let myBooks = null;
+
+function makeGrey(book, index) {
+  const isGrey = book.node.classList.contains('grey');
+  if (index % 2 === 0 && !isGrey) {
+    book.node.classList.add('grey');
+  } else if (isGrey && index % 2 !== 0) {
+    book.node.classList.remove('grey');
+  }
+}
+
+const createBooksElements = (book) => {
+  const div = document.createElement('div');
+
+  const info = document.createElement('p');
+  info.innerText = `"${book.title}"\tby\t${book.name}`;
+
+  const remove = document.createElement('button');
+  remove.classList.contains('grey');
+  remove.innerHTML = 'Remove';
+  remove.addEventListener('click', () => {
+    myBooks.remove(book.id, div);
+  });
+
+  div.append(info, remove);
+  Bookscontainer.appendChild(div);
+
+  return div;
+};
```
- [x] Based on the requirements all functions should be refactored to classes.

FrederickMih commented on script.js:
```diff
@@ -0,0 +1,98 @@
+    }
+}
+
+function Book(id, title, name) {
+  this.id = id;
+  this.title = title;
+  this.name = name;
+  this.node = null;
+}
+
+Form.onsubmit = (e) => {
+  e.preventDefault();
+  myBooks.add(new Book(myBooks.count, Form.title.value, Form.name.value));
+  Form.title.value = '';
+  Form.name.value = '';
+};
+
+window.addEventListener('DOMContentLoaded', () => {
+  myBooks = new Books();
+  const objects = localStorage.getItem('books').split('|');
+
+  if (objects[0] === '') { return; }
+
+  myBooks.array = objects.map((object) => JSON.parse(object));
+
+  myBooks.array.forEach((book, index) => {
+    const node = createBooksElements(book);
+    book.node = node;
+    makeGrey(book, index);
+  });
+});
```
- [x] As pointed out above :arrow_up: let's refactor functions to classes.

FrederickMih commented on script.js:
```diff
@@ -0,0 +1,98 @@
+
+    remove(i) {
+      Bookscontainer.removeChild(this.array[i].node);
+      this.array.splice(i, 1);
```
- [**Optional**] Let's favor the use of array methods that don't mutate array like [concat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat), [filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter), [slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) instead of [push](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push), [pop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/pop), [splice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice)

FrederickMih reviewed (changes requested):
## Hi :raising_hand_man:  @tsohleDev , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :green_circle: 

- App styles match the given wireframe :+1: 
- Descriptive PR :+1: 


## Required Changes ♻️
N/A
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

tsohleDev committed:
Add modules

tsohleDev committed:
Remove functionality from script.js

Remove the functions under form.submit and windows.Domcontentloaded
and encapsulate them in Component class and Books class

ShadyShawkat reviewed (approved):
## Hi @tsohleDev, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All linters are passing ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

![Image](https://media.giphy.com/media/jJQC2puVZpTMO4vUs0/giphy.gif)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
